.github/workflows/build.yml

```
@@ -8,7 +8,13 @@ on:
   push:
     branches:
       - main
+    paths:
+      - common/**
+      - fabric/**
   pull_request:
+    paths:
+      - common/**
+      - fabric/**
```

Reuse paths list using &\* syntax, and add other files referenced during build (e.g. the flake)
```suggestion
    paths: &paths
      - common/**
      - fabric/**
      - build.gradle
      - gradle.properties
      - flake.nix
      - flake.lock
      - .github/workflows/build.yml
  pull_request:
    paths: *paths
```